### ISSUE TITLE
Remove empty lines

### ISSUE THREAD

**Issue by demisx** [lazd/gulp-replace#74]
Is it possible to remove empty lines in files using `gulp-replace`? I've tried this, but it didn't work:

```js
  .pipe(replace(/^\s*$/, '', { skipBinary: true }))
```

Thanks.

**Comment by lazd** [comment]
Hi @demisx, this is a regex question, try posting it on Stack Overflow and you're sure to get some answers!

**Comment by lazd** [comment]
If you do find you're unable to get it to work after trying other strategies, please re-open the issue.

**Comment by demisx** [comment]
Hi. Thank you for the quick response. I know that `/^\s*$/` regex matches an empty line. It just doesn't work with `gulp-replace` for some reason. I'll try `gulp-remove-empty-lines` instead.

**Comment by lazd** [comment]
@demisx Interesting, I will investigate it.

**Comment by lazd** [comment]
@demisx that matches the whitespace itself, but it doesn't match the newline. Did you try `/^\s*\n/` or something like that?

**Comment by demisx** [comment]
No, I don't believe I've tried this one yet.  And it did work. Sorry for the false alarm -- I need to refresh my memory on regex matches. Stupid me. Totally forgot that `$` doesn't match `\n`.

**Comment by lazd** [comment]
@demisx it happens to the best of us, I'm glad you solved it!

In the mean time, I am adding a test for this because it IS possible there could be weirdness between a buffered and streamed replace involving a newline, and I want to make sure it's working. Thanks!

**Comment by demisx** [comment]
@lazd You are awesome man! I wish I could give you more than one star.

**Comment by lazd** [comment]
@demisx so apparently `/^\s*\n/gm` is the right way to do it, or `/^\s*[\r\n]/gm` to also catch legacy Mac line endings. You need the `m` flag to match on a per-line basis, otherwise `^` matches the start of the string.

That said, the results are inconsistent between streams and buffers! This regex does not work when input is buffered, so this is still an issue!

**Comment by lazd** [comment]
@demisx I pushed some tests to the `blanklines` branch, maybe you can have a look and see why the buffered test is failing? That's better than a hundred stars!

**Comment by dashawk** [comment]
tried it and it works.
```js
replace(/(?:(?:\r\n|\r|\n)\s){2}/gm, '\r\n')
```

**Comment by revelt** [comment]
Correct me if I'm wrong but the Sept 21st 2016 comment contains `/^\s*[\r\n]/gm` which might be incorrect. Brackets means "one of these characters". If there is `\r\n` this would catch `\r` and not `\n`.

If you want to check in [regexr](https://regexr.com/), try: `\n[ab]` with `gm` flags and following copy:

```

ab


```

![regex](https://user-images.githubusercontent.com/8344688/40185428-7de0eb1e-59ea-11e8-9597-adc6b3a9cbc4.png)

@dashawk snippet is better, it uses "or" but it only removes more than 2 consecutive empty lines. I'm not sure that's exactly what was asked — cases of more than 1 consecutive empty line should be replaced with a single line break.

---

What if we simply replaced instances of:

```
/(\r\n|\r|\n){2,}/gm
```

with `\n`? The `{2,}` bit means "two or more".
